sweet/benchmarks/cockroachdb/main.go

```diff
@@ -0,0 +1,611 @@
+					time.Sleep(5 * time.Second)
+					if err := inst.ping(cfg); err == nil {
+						cancel()
+						return
+					}
+				}
+			}
+		}(ctx)
+	}
+
+	select {
+	case <-ctx.Done():
+	case <-time.After(time.Minute):
+		return errors.New("benchmark timed out waiting for cluster to be ready")
+	}
+
+	return nil
+}
+
+func (i *cockroachdbInstance) ping(cfg *config) error {
+	// Ping the node to see if it is live. The actual command of
+	// `node status` is a bit arbitrary, if it responds at all
+	// we know the node is live. Generally it is used to see if
+	// *other* nodes are live.
+	cmd := exec.Command(cfg.cockroachdbBin,
+		"node",
+		"status",
+		"--insecure",
+		fmt.Sprintf("--host=%s", cfg.host),
+		fmt.Sprintf("--port=%d", i.sqlPort),
+	)
+	cmd.Stdout = &i.output
+	cmd.Stderr = &i.output
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (i *cockroachdbInstance) sqlAddr() string {
+	return fmt.Sprintf("%s:%d", cliCfg.host, i.sqlPort)
+}
+
+func (i *cockroachdbInstance) httpAddr() string {
+	return fmt.Sprintf("%s:%d", cliCfg.host, i.httpPort)
+}
+
+func (i *cockroachdbInstance) shutdown() error {
+	// Only attempt to shut down the process if we got to a point
+	// that a command was constructed and started.
+	if i.cmd != nil && i.cmd.Process != nil {
+		// Send SIGTERM and wait instead of just killing as sending SIGKILL
+		// bypasses node shutdown logic and will leave the node in a bad state.
+		// Normally not an issue unless you want to restart the cluster i.e.
+		// to poke around and see what's wrong.
+		if err := i.cmd.Process.Signal(syscall.SIGTERM); err != nil {
+			return err
+		}
+		if _, err := i.cmd.Process.Wait(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type benchmark struct {
+	name        string
+	reportName  string
+	workload    string
+	nodeCount   int
+	args        []string
+	longArgs    []string // if !config.short
+	shortArgs   []string // if config.short
+	metricTypes []string
+	timeout     time.Duration
+}
+
+const (
+	readMetric  = "read"
+	writeMetric = "write"
+)
+
+func kvBenchmark(readPercent int, nodeCount int) benchmark {
+	metricTypes := []string{writeMetric}
+	if readPercent > 0 {
+		metricTypes = append(metricTypes, readMetric)
+	}
+
+	return benchmark{
+		name:        fmt.Sprintf("kv%d/nodes=%d", readPercent, nodeCount),
+		reportName:  fmt.Sprintf("CockroachDBkv%d/nodes=%d", readPercent, nodeCount),
+		workload:    "kv",
+		nodeCount:   1,
+		metricTypes: metricTypes,
+		// Very generous timeout, we don't expect to ever hit this, but just in case.
+		timeout: 5 * time.Minute,
+		args: []string{
+			"workload", "run", "kv",
+			fmt.Sprintf("--read-percent=%d", readPercent),
+			"--min-block-bytes=1024",
+			"--max-block-bytes=1024",
+			"--concurrency=10000",
+			"--max-rate=30000",
+		},
+		longArgs: []string{
+			"--ramp=15s",
+			"--duration=1m",
+		},
+		// Chosen through trial and error as the shortest time that doesn't
```

Comment:
The results of short runs may be more stable if we pre-split and scatter the kv table before the test run.

Reply:
Done! That might've helped? At the very least it didn't seem to make it worse so I'll keep it in there.

Question on the value to set `--splits` to though. I set it to `5` because that's how many ranges there were at the end without pre-splitting. Is that reasonable? Seems low to me but it shouldn't be much data?